pkg/reportingplugins/mercury/fees.go

```diff
@@ -1,24 +1,41 @@
 package mercury
 
-import "math/big"
+import (
+	"github.com/pkg/errors"
+	"math/big"
+)
 
 // PriceScalingFactor indicates the multiplier applied to token prices.
 // e.g. for a 1e8 multiplier, a LINK/USD value of 7.42 will be represented as 742000000
 // This is what we expect from our data source.
-var PRICE_SCALING_FACTOR = big.NewInt(1e8)
+var PRICE_SCALING_FACTOR = big.NewFloat(1e8)
 
 // FeeScalingFactor indicates the multiplier applied to fees.
 // e.g. for a 1e18 multiplier, a LINK fee of 7.42 will be represented as 7.42e18
 // This is what will be baked into the report for use on-chain.
-var FEE_SCALING_FACTOR = big.NewInt(1e18)
+var FEE_SCALING_FACTOR = big.NewFloat(1e18)
 
-var CENTS_PER_DOLLAR = big.NewInt(100)
+var CENTS_PER_DOLLAR = big.NewFloat(100)
 
 // CalculateFee outputs a fee in wei
-func CalculateFee(tokenPriceInUSD *big.Int, baseUSDFeeCents uint32) (fee *big.Int) {
-	fee = new(big.Int).Mul(big.NewInt(int64(baseUSDFeeCents)), tokenPriceInUSD)
-	fee = fee.Mul(fee, FEE_SCALING_FACTOR)
-	fee = fee.Div(fee, PRICE_SCALING_FACTOR)
-	fee = fee.Div(fee, CENTS_PER_DOLLAR)
-	return
+func CalculateFee(tokenPriceInUSD *big.Int, baseUSDFeeCents uint32) (*big.Int, error) {
+	if tokenPriceInUSD.Cmp(big.NewInt(0)) == 0 || baseUSDFeeCents == 0 {
+		return nil, errors.Errorf("token price and base fee must be non-zero")
+	}
+
+	// big.Float base fee in USD
+	baseFee := new(big.Float).Quo(new(big.Float).SetInt64(int64(baseUSDFeeCents)), CENTS_PER_DOLLAR)
```

Comment:
Isn't cents_per_dollar and price_scaling_factor the same thing? it looks like you need to do something like:

baseUSDFeeCents * token.scale / tokenPriceInUsd

Reply:
PRICE_SCALING_FACTOR is the multiplier applied to the price we retrieve from reports. 
CENTS_PER_DOLLAR is just to convert fee in cents that we receive from config into usd.

Reply:
I believe you only need one, and that is the precision of the token